wit-0.3.0-draft/stdio.wit

```diff
@@ -1,17 +1,59 @@
+@since(version = 0.3.0-rc-2025-08-15)
+interface stdio {
+  @since(version = 0.3.0-rc-2025-08-15)
+  enum error-code {
+    /// Input/output error
+    io,
+    /// Invalid or incomplete multibyte or wide character
+    illegal-byte-sequence,
+    /// Broken pipe
+    pipe,
+  }
+}
+
 @since(version = 0.3.0-rc-2025-08-15)
 interface stdin {
+  /// Return a stream for reading from stdin.
+  ///
+  /// This function returns a stream which provides data read from stdin,
+  /// and a future to signal read results.
+  ///
+  /// If the stream's readable end is dropped the future will resolve to success.
+  ///
+  /// If the stream's writable end is dropped the future will either resolve to
+  /// success if stdin was closed by the writer or to an error-code if reading
+  /// failed for some other reason.
+  ///
+  /// Multiple streams may be active at the same time. The behavior of concurrent
+  /// reads is implementation-specific.
   @since(version = 0.3.0-rc-2025-08-15)
-  get-stdin: func() -> stream<u8>;
+  read-stdin: func() -> tuple<stream<u8>, future<result<_, error-code>>>;
 }
 
 @since(version = 0.3.0-rc-2025-08-15)
 interface stdout {
+  /// Append from the given stream to stdout.
+  ///
+  /// If the stream's writable end is dropped this function will either return
+  /// success once the entire contents of the stream have been written or an
+  /// error-code representing a failure.
+  ///
+  /// Otherwise if there is an error the readable end of the stream will be
+  /// dropped and this function will return an error-code.
   @since(version = 0.3.0-rc-2025-08-15)
-  set-stdout: func(data: stream<u8>);
+  append-stdout: async func(data: stream<u8>) -> result<_, error-code>;
```

Comment:
The word "append" here feels counterintuitive to me. What would you think of calling it `write-via-stream`, similar to wasi-filesystem's `write-via-stream`? And similar for stderr. And stdin could use `read-via-stream`.